base/logging.jl

```diff
@@ -385,7 +385,7 @@ function logmsg_code(_module, file, line, level, message, exs...)
                         end
                         line = $(log_data._line)
                         local msg, kwargs
-                        $(logrecord) && handle_message(
+                        $(logrecord) && invokelatest(handle_message,
```

Comment:
I guess this could also be controversial. Fundamentally I don't think it's good if the logging code invalidates the code it gets embedded in. And despite my best efforts the logging code is extremely susceptible to invalidation (open-ended flexibility or invalidation-resistant, pick one). But in principle this could cause a slowdown for `@debug` statements.

Reply:
`invokelatest` is equally fast as a dynamic dispatch, it just makes it explicit that is intended

Reply:
Probably `_invoked_shouldlog` should also do the same?

Reply:
SGTM, I will add that.

There may be cases where everything is inferrable and runtime dispatch wouldn't be necessary, but the `@nospecialize`s kind of obviate the advantage anyway.